clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -1372,11 +1372,13 @@ public String toEnrichedRst() {
      */
     private void getConfigKeyRst(ConfigKey key, StringBuilder b) {
         b.append("``").append(key.name).append("``").append("\n");
-        for (String docLine : key.documentation.split("\n")) {
-            if (docLine.length() == 0) {
-                continue;
+        if (null != key.documentation) {
+            for (String docLine : key.documentation.split("\n")) {
+                if (docLine.length() == 0) {
+                    continue;
+                }
+                b.append("  ").append(docLine).append("\n\n");
```

Comment:
It seems `toHTLM()` is not tested but there's an existing test for `toEnrichedRst()` in `ConfigDefTest`. Can we add a config entry with a `null` documentation there?

What about something like:
```
diff --git clients/src/test/java/org/apache/kafka/common/config/ConfigDefTest.java clients/src/test/java/org/apache/kafka/common/config/ConfigDefTest.java
index f8239e5b7e..5a4655db14 100644
--- clients/src/test/java/org/apache/kafka/common/config/ConfigDefTest.java
+++ clients/src/test/java/org/apache/kafka/common/config/ConfigDefTest.java
@@ -531,9 +531,15 @@ public class ConfigDefTest {
                         "Group Two", 1, Width.NONE, "..", Collections.<String>emptyList())
                 .define("opt1.of.group2", Type.BOOLEAN, false, Importance.HIGH, "Doc doc doc doc doc.",
                         "Group Two", 0, Width.NONE, "..", Collections.singletonList("some.option"))
-                .define("poor.opt", Type.STRING, "foo", Importance.HIGH, "Doc doc doc doc.");
+                .define("poor.opt", Type.STRING, "foo", Importance.HIGH, "Doc doc doc doc.")
+                .define("null doc", Type.STRING, "foo", Importance.HIGH, null);

         final String expectedRst = "" +
+                "``null doc``\n" +
+                "  * Type: string\n" +
+                "  * Default: foo\n" +
+                "  * Importance: high\n" +
+                "\n" +
```

Reply:
Hello @mimaison! Thanks for your review!
I applied the changes and added some code for it to the test cases.
I also discovered an issue with the RST output by adding the test!